### PR TITLE
CARMA Basic Travel UI - Initial Implementation - Layout

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
This story shall concentrate on the initial implementation of the CARMA Basic Travel UI which involves the layout and project setup. 
## Description
The developer needs to create a stand alone branch from integration/routing branch. 
Integration/routing branch may have build issues therefore need to work with team members to resolve any issues that affects this user story. If not doable within the timeframe, document the issues in Github to be resolved and prioritized. 
Login and Logout should be working order, including the desktop icon link, similar to CARMA Messenger layout.
The layout should cover the header, footer and display panels. 
Route selection and display should be working. 
Plugin display and selection should be working which includes displaying required plugins. Note will only assume what already has been implemented on Guidance. 
This assumes that some of the topics, services, plugins and other capabilities may not be implemented or working order. Therefore developer could choose to dummy the data or have a placeholder at this time to display the new layout with the graphics and icons, which means some actual functionality may not work at the time. 

UI look and feel changes according to UI mock design
project folder structure modifications
Reimplemented ros connection and ros topics
## How Has This Been Tested?
Launch UI and backend ros services
Complete integration testing

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by maefromm** [comment]
@dan-du-car The "backend_services/2D" 
1) Thinking this should probably be under scripts since the rosbridge.js is still intact which will confuse users. 
2) widget_utils and widgets, what is the difference between the two? Couldn't you just have it under the widgets folder instead of having this top level folder? 
Need to further discuss today on the new folder structure now that you've started to develop the layout more.

**Comment by dan-du-car** [comment]
> @dan-du-car
> 
> 1. "backend_services/2D", thinking this should probably be under scripts since the rosbridge.js is still intact which will confuse users.
> 2. widget_utils and widgets, what is the difference between the two? Couldn't you just have it under the widgets folder instead of having this top level folder?
>    Need to further discuss today on the new folder structure now that you've started to develop the layout more.

@maefromm 
1. rosbridge.js was the old version, it is temporary left there for now. It will be removed once the whole UI implementation is complete.
2. /Widget_utils contains the widgets' look and feel, and JS functionalities. It does not have ROS related functionalities like /widgets folder does. /Widgets_utils will be used by the /widgets folder down the road implementing ROS functionalities for widgets.The idea here is to separate the ROS services/topics from client side JS and css. /widget_utils functions are more like controls.js. 
> Letting me know when you are available for discussion today.

**Comment by maefromm** [comment]
1) Add more space below, and maybe make the button just a bit bigger. 
image
2) Is the Statistics and Advanced supposed to show a menu list? Causing a bootstrap dropdown error for me. 
image
3) Please update to just say "Are you sure you want to take manual control? NOTE we will need to discuss the flow of the UI popups once i'm done with the UI design state. 
image
4) You had a different marker for Pinpoint GPS, do you want to use that instead? I'm ok either way.
image
5) OBU status is pending Laurie's input. 
6) For the status icons, please go ahead and put the titles for each icon that you suggested and we'll have to ensure it works on touch. Please center the text to the icon. 
7) You've updated Plugins to Active plugins, this is ok for now. Need to check the topic that you'll get this from. 
image

**Comment by dan-du-car** [comment]
> 1. Add more space below, and maybe make the button just a bit bigger.
>    image
> 2. Is the Statistics and Advanced supposed to show a menu list? Causing a bootstrap dropdown error for me.
>    image
> 3. Please update to just say "Are you sure you want to take manual control? NOTE we will need to discuss the flow of the UI popups once i'm done with the UI design state.

>    image
> 4. You had a different marker for Pinpoint GPS, do you want to use that instead? I'm ok either way.
>    image
> 5. OBU status is pending Laurie's input.
> 6. For the status icons, please go ahead and put the titles for each icon that you suggested and we'll have to ensure it works on touch. Please center the text to the icon.
> 7. You've updated Plugins to Active plugins, this is ok for now. Need to check the topic that you'll get this from.
>    image

@maefromm 
1. Sure. Will do.
2. Please make sure you have this within your header file in main.html:
 <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
 3. Sure will do.
4. Will change to your icon, please send me the icon you mentioned above. But I will leave the GPS icon change to next task because it is part of the next task. 
5. Sure.
6. The title is already there. 
7. It is getting from  topic: /guidance/plugins/get_active_plugins

**Comment by maefromm** [comment]
@dan-du-car I still have to run the images, haven't gotten around to it yet, so maybe to come in terms of PR comments.

File: website/thirdparty/fontawesome-free-5.13.1-web/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Font Awesome Free License
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/thirdparty/fontawesome-free-5.13.1-web/LICENSE.txt]: Addressed

File: website/index.html [hunk 2]
```diff
@@ -1,107 +1,142 @@
...
+                       </div>                                  
+                    </div>
+                
+                    <!--  Carma Options Description/Help Info Modal -->
+                    <div class="modal fade" id="CarmaOptionsModal" tabindex="-1" role="dialog" aria-labelledby="CarmaOptionsModalTitle" aria-hidden="true">
+                        <div class="modal-dialog modal-dialog-centered modal-dialog modal-lg" role="document">
+                        <div class="modal-content">
+                            <div class="modal-header">
+                                <h4 class="modal-title" id="exampleModalLongTitle"  style="text-align: left;">
+                                    <strong>Platform Options</strong>
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/index.html]: addressed

File: website/index.html [hunk 3]
```diff
@@ -1,107 +1,142 @@
...
+                                </h4>
+                                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                                    <span aria-hidden="true">&times;</span>
+                                </button>
+                            </div>
+                            <div class="modal-body" style="text-align: left;"> 
+                                <p>"Debug Mode" provides the option to allow users to view CARMA platform's internal state.</p>
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/index.html]: addressed

File: website/main_backup.html [hunk 4]
```diff
@@ -0,0 +1,261 @@
+﻿<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/main_backup.html]: addressed

File: website/index_backup.html [hunk 5]
```diff
@@ -0,0 +1,107 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/index_backup.html]: b3067ce70e691a1e3c0adfddb1858bc21727bd50

File: website/main_backup.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/main_backup.html]: b3067ce70e691a1e3c0adfddb1858bc21727bd50

File: website/index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/index.html]: b3067ce70e691a1e3c0adfddb1858bc21727bd50

File: website/thirdparty/fontawesome-free-5.13.1-web/LICENSE.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/thirdparty/fontawesome-free-5.13.1-web/LICENSE.txt]: b3067ce70e691a1e3c0adfddb1858bc21727bd50

File: website/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (dan-du-car) [reply website/index.html]: b3067ce70e691a1e3c0adfddb1858bc21727bd50

**Comment by maefromm** [comment]
@dan-du-car missing "Login and Logout should be working order, **including the desktop icon link**, similar to CARMA Messenger layout." Please follow the similar installation script and update the desktop icon to be CARMA Platform icon to be installed.

**Comment by maefromm** [comment]
@dan-du-car "Debug Mode" actually should say "... provides the option to **not** launch the backend services when clicking the "Start" button. " Please set that to Off by Default.

**Comment by maefromm** [comment]
@dan-du-car got this error when carma start all and chose to disable the remote start (Debug Mode = Off). 
image

**Comment by maefromm** [comment]
@dan-du-car The disengage button here is not needed, since the C-ADS carma button will be the dis-engage button. 
image

**Comment by maefromm** [comment]
@dan-du-car not sure if you meant to check in the png for this
image

File: website/test/html/ChangePluginsSample.html [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                  <div class="card-header card-custom-header" style="text-align: center;padding-left: 0px;">
+                      <button type="button" class="btn btn-info" id="changePluginsCloseBtn">X</button>
+                      <div>
+                         Change Plugins
+                      </div> 
+                  </div>
+                  <div class="card-body" >
+                    <p class="card-text warning-messages">
+                        <i class="fas fa-exclamation-triangle"></i>
+                        <span>Please proceed with caution, any changes to these plugins will change vehicle behavior</span>
```
Comment (maefromm) [inline website/test/html/ChangePluginsSample.html:19]: Please update to : 
"Please proceed with caution. Any changes to these plugins will change vehicle behavior."

**Comment by maefromm** [comment]
As discussed, all the other changes look good for now. Just need to fix the login error mentioned above. 
Once that's done, 
1) please create a new integration/ui from integration/routing.
2) Update this PR to merge into that. Please be careful not to delete the comments we made in the PR.

File: website/test/html/ChangePluginsSample.html (hunk 6 above)
Comment (dan-du-car) [reply website/test/html/ChangePluginsSample.html:19]: https://github.com/usdot-fhwa-stol/carma-web-ui/commit/3b75b3d4bd09d69cb18eac69579c8e5015d802cb

**Comment by maefromm** [comment]
@dan-du-car as discussed, the error was due to the carma-config repo's carma.config.js script being old so manually had to update. The Debug Mode worked but the remote did not. You mentioned yours worked so I trust it's just my setup that' the issue. Please go ahead and update this PR to merge into the new "integration/ui" branch, instead of the "develop" branch.

**Comment by dan-du-car** [comment]
> As discussed, all the other changes look good for now. Just need to fix the login error mentioned above.
> Once that's done,
> 
> 1. please create a new integration/ui from integration/routing.
> 2. Update this PR to merge into that. Please be careful not to delete the comments we made in the PR.

Addressed

**Pull request merged**